### ISSUE TITLE
RememberMe and GoogleAuth verification bug

### ISSUE THREAD

**Issue by anvyst** [CakeDC/users#488]
The bug spotted with RememberMe and GoogleAuthenticator functionality.
If normal session expires and RememberMe is enabled GoogleAuthenticator still requires you to verify the user credentials, however is should only check if RememberMe isn't expired and and do a normal relogin without GoogleAuthenticator verification.

Apparently and extra check for RememberMe cookie should be added here:
```php
//src/Controller/Traits/LoginTrait.php 
//method: _afterIdentifyUser()

if ($googleAuthenticatorLogin) {                                                                                                           
                $url = Configure::read('GoogleAuthenticator.verifyAction');                                                                            
                                                                                                                                                       
                return $this->redirect($url);                                                                                                          
            }      
....
```
By default, `remember_me` cookie is set no matter whether the checkbox is clicked or not. This can fix the issue:

```php
//src/Controller/Component/RememberMeComponent.php
//method: setLoginCookie()
if (!(bool)$event->subject()->request->data('remember_me')) {
            return;
        }
```

**Comment by steinkel** [comment]
I think we are OK to close this one, right @anvyst ?

**Comment by anvyst** [comment]
@steinkel , 

The issue still remains and it references to #490 at some extend.
If you have `RememberMeComponent` enabled in the config, `/login` doesn't pay attention whether 'Remember Me' checkbox is enabled or not - RememberMe cookie will be set in any case.
If we disable the component the plugin breaks as it cannot find the component listed in the `Auth.authenticate` array of the config. (so it might be added automagically in the config list based on `Users.RememberMe.active` flag).

Uncontrollable loading of RememberMe component (as mentioned above) by the plugin creates an issue on what to do with the `/verify` action. When the user is logged in and verified by GAuthenticator it will be kicked out of the system after user session expires, and rememberMe cookie won't let him in - you'll have to verify the credentials again. 

We did a temporary solution on our side - disabled RememberMe on app level and stretched user session timeout which not really elegant.

Ideally: 
1. If `Users.RememberMe.active` is enabled - we should instantiate RememberMe cookie and add it in `config` array of authenticators.
2. In case of GoogleAuthenticator the cookie should be inited after successful `/verify` action. If not - after successful `/login`.

**Comment by steinkel** [comment]
closed per #599 please let me know if that's still an issue
